tensorflow_datasets/image/tiered_imagenet.py

```diff
@@ -0,0 +1,139 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl import logging
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+import io
+import os
+import numpy as np
+import pickle as pkl
+from tqdm import tqdm
+import cv2
+
+
+class TieredImagenet(tfds.core.GeneratorBasedBuilder):
+    """tiered_imagenet is comparatively larger subset of Imagenet that uses splits defined for benchmarking few-shot learning approaches"""
+    _CITATION = """\
+            @inproceedings{ren18fewshotssl,
+            author   = {Mengye Ren and
+                        Eleni Triantafillou and
+                        Sachin Ravi and
+                        Jake Snell and
+                        Kevin Swersky and
+                        Joshua B. Tenenbaum and
+                        Hugo Larochelle and
+                        Richard S. Zemel},
+            title    = {Meta-Learning for Semi-Supervised Few-Shot Classification},
+            booktitle= {Proceedings of 6th International Conference on Learning Representations {ICLR}},
+            year     = {2018},
+            }
+        """
+    _DESCRIPTION = """\
+            The tieredImageNet is a dataset proposed specifically for few-shot classification. 
+            Like miniImagenet, it is subset of ILSVRC-12, but larger subset with total 608 classes. 
+            The dataset is proposed to make use of class hierarchy while few-shots classficicatio.
+            It is analogous to Omniglot dataset where the characters are grouped into alphabets.
+            tieredImageNet groups classes into broader categories that corresponds to higher-level nodes 
+            in the ImageNet hierarchy.
+            tieredImageNet has total 34 general categories. Each category containing between 10 and 30 
+            classes. The categories are split into 20 training, 6 validation and 8 testing categories.
+            Each category has corresponding more specific classes.
+        """
+
+    _BASE_URL = "https://github.com/renmengye/few-shot-ssl-public.git"

```

Comment:
Maybe `arxiv` links better than that. Because we need some explanation about dataset.

Reply:
added.